### PR TITLE
WT-7649 Skip timestamp assert during recovery

### PR DESCRIPTION
**Pull request by alisonfel**
We don't expect recovery to be using timestamps when applying commits. If a table is configured with write timestamp asserts, being `assert=(write_timestamp=on),write_timestamp_usage=always)`, the recovery process will trigger asserts and fail. To avoid failing/blocking recovery, updated the timestamp assert to be skipped on a recovery connection.

### CODE REVIEW

File: src/txn/txn.c [hunk 1]
```diff
@@ -1257,15 +1258,20 @@ __txn_commit_timestamps_assert(WT_SESSION_IMPL *session)
...
     used_ts = F_ISSET(txn, WT_TXN_HAS_TS_COMMIT) || F_ISSET(txn, WT_TXN_HAS_TS_DURABLE);
+    recovery_conn = F_ISSET(conn, WT_CONN_RECOVERING);
     /*
-     * Debugging checks on timestamps, if user requested them.
+     * Debugging checks on timestamps, if user requested them. We additionally don't expect recovery
+     * to be using timestamps when applying commits. If recovery is running, skip this assert to
+     * avoid failing the recovery process.
      */
-    if (F_ISSET(txn, WT_TXN_TS_WRITE_ALWAYS) && !used_ts && txn->mod_count != 0)
+    if (F_ISSET(txn, WT_TXN_TS_WRITE_ALWAYS) && !used_ts && txn->mod_count != 0 && !recovery_conn)
```
Comment (sueloverso) [inline src/txn/txn.c:1274]: You're only using both local variables once. I would simplify it by removing those and making the conditional as 
`!F_ISSET(S2C(session), WT_CONN_RECOVERING)`.

**Review by sueloverso — APPROVED** [review]
After simplifying, LGTM.

File: src/txn/txn.c (hunk 1 above)
Comment (alisonfel) [reply src/txn/txn.c]: Good point! I've removed the local variable and added `!F_ISSET(S2C(session), WT_CONN_RECOVERING)` into the conditional.

**Review by tetsuo-cpp — APPROVED** [review]
LGTM!

**Pull request merged**